core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -119,6 +125,10 @@ public synchronized ChannelImpl shutdown() {
       return this;
     }
     shutdown = true;
+    if (deadlineCancellationExecutor != null) {
+      deadlineCancellationExecutor =
```

Comment:
Ah, because `ExecuteExistingDelayedTasksAfterShutdownPolicy` defaults to `true`.

Yes, if `shutdown`, we will call `closeCallPrematurely()` because of `obtainActiveTransport()`'s return value.